app/src/main/res/layout/activity_product.xml

```diff
@@ -40,14 +39,27 @@
         android:id="@+id/buttonScan"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
-        android:layout_alignParentBottom="true"
-        android:layout_alignParentEnd="true"
-        android:layout_alignParentRight="true"
-        android:layout_gravity="top|center_horizontal"
-        android:layout_margin="@dimen/activity_vertical_margin"
+        app:layout_anchor ="@id/pager"
+        app:useCompatPadding="true"
+        app:fabSize="normal"
+        app:elevation="10dp"
+        app:layout_anchorGravity = "bottom|right|end"
```

Comment:
You  should be able to anchor this to the top of the navigation bar. We want the floating action button "sitting" on top of the navigation bar, half inside it and half outside it. Have a look at the [top answer](https://stackoverflow.com/questions/37384957/floatingactionbutton-anchor-on-top-of-a-view) here.